### PR TITLE
chore: Setup automated semver bumps

### PR DESCRIPTION
**Pull request by tssweeney**

### CODE REVIEW

File: weave/version.py
Comment (andrewtruong) [inline weave/version.py:1]: Why do we have this file?  Shouldn't version just be in `pyproject.toml`?

Comment (tssweeney) [reply weave/version.py:1]: This is used to have the version available at runtime

Comment (andrewtruong) [reply weave/version.py:1]: Can't you use `weave.__version__`?

Comment (tssweeney) [reply weave/version.py:1]: weave.__version__ IS defined by `version.py`: 
Screenshot 2024-08-16 at 14 12 15

Comment (andrewtruong) [reply weave/version.py:1]: hmm, I was expecting something more like this:
https://stackoverflow.com/a/67097076

Comment (tssweeney) [reply weave/version.py:1]: This works, i would like to use it as is (:

**Review by andrewtruong — APPROVED** [review]

**Pull request merged**